### PR TITLE
fix: Incorrect typing in samples/cores/loop_parallism and fixing loop_parameter examples

### PR DESCRIPTION
**Pull request by papagala**
**Description of your changes:**

The type hint on `loop_parallelism` is incorrect, leading to a failure in compiling with the latest version of `kfp` Version: 2.8.0. Without this MR you get the following error:

```bash
Traceback (most recent call last):
...
    raise InconsistentTypeException(error_text)
kfp.dsl.types.type_utils.InconsistentTypeException: Incompatible argument passed to the input 's' of component 'print-op': Argument type 'NUMBER_INTEGER' is incompatible with the input type 'STRING'
```

I also took the time to fix `loop_parameter.py` 

It was throwing this error without the changes applied in this MR

```python
    raise ValueError(
ValueError: Cannot iterate over a single parameter using `dsl.ParallelFor`. Expected a list of parameters as argument to `items`.
```

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by gregsheremeta** [comment]
/ok-to-test

Thanks for the PR!

**Comment by HumairAK** [comment]
/lgtm
/approve

@chensun

**Comment by papagala** [comment]
@gregsheremeta I added another fix. I will wait until this is merged to add any additional ones :)

**Comment by papagala** [comment]
Thanks a lot @hbelmiro

**Comment by gregsheremeta** [comment]
> I added another fix. I will wait until this is merged to add any additional ones :)

thanks! One per PR is probably good.

If you want to leave the [loop_parameter.py](https://github.com/kubeflow/pipelines/pull/11062/files#diff-678b1a8969b80760721d4cf831794d8ac719ec4f4bcb17bda4e62114341ebdeb) fix in here, just be sure to update the PR title and message to reflect what's being fixed there too.

**Comment by papagala** [comment]
@gregsheremeta I have changed the title and description :)

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks for the fix!

**Comment by papagala** [comment]
> /lgtm /approve
> 
> Thanks for the fix!

It is imy pleasure. I'll try to fix some other samples :)

**Pull request merged**
